EIPS/eip-6963.md

```diff
@@ -158,25 +158,40 @@ interface EIP6963AnnounceProviderEvent extends CustomEvent {
 
 Therefore both wallet providers and the Ethereum library must emit events when they initialize.
 
+Page script listens to the `"eip6963:announceProvider"` event and dispatches the `"eip6963:requestProvider"` event:
 ```typescript
-// Ethereum library initializes
-window.dispatchEvent(new Event("eip6963:requestProvider"));
+// Page script listens to announced providers
+window.addEventListener(
+  "eip6963:announceProvider",
+  (event: EIP6963AnnounceProviderEvent) => {}
+);
 
-// Wallet provider initializes
-window.dispatchEvent(new CustomEvent("eip6963:announceProvider", { detail }));
+// Page script dispatches a request event which will be heard by 
+// content scripts injected earlier
+window.dispatchEvent(new Event("eip6963:requestProvider"));
 ```
 
-Additionally the Wallet providers will react to the event emitted by the Ethereum library.
+Content script listens to the `"eip6963:requestProvider"` event and dispatches the `"eip6963:announceProvider"`:
 
 ```typescript
-window.addEventListener(
-  "eip6963:requestProvider",
-  (event: EIP6963RequestProviderEvent) => {
-    announceProvider();
-  }
+const announceEvent: EIP6963AnnounceProviderEvent = new Event(
```

Comment:
This should be a `CustomEvent` as there is a `detail` field.

Reply:
Right, good catch! Thanks